### PR TITLE
[camera] Changed the order of the setters for mediaRecorder in MediaRecorderBu…

### PR DESCRIPTION
**Pull request by valconetti**
…ilder.java to make it more readable.

## Description

The `mediaRecorder` expects the setters in a specific order. This order was however not ideal. Since there is some wiggle room I changed up the order a bit so the audio setters are in the same if statement where possible. I also added the link to the documentation about it.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java
Comment (amirh) [inline packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java:53]: Thanks for the link! it makes more sense now.

nit: it doesn't seem like the requirement is for a specific fixed ordering but that in each media recorder state a different set of calls are allowed (e.g we can replace the ordering for setAudioSource and setVideoSource).

Maybe the comment should say something like:

```
The setters below drive the internal state machine of the media recorder, these setters cannot be called in any ordering but must satisfy the state constraints detailed in: https://developer.android.com/reference/android/media/MediaRecorder.
```

**Review by amirh — APPROVED** [review]
LGTM modulo comment nit

**Comment by amirh** [comment]
Thanks! can you rebase?

**Comment by mvanbeusekom** [comment]
Merged as build on Cirrus is green.

**Comment by amirh** [comment]
Note that the tree was red during the weekend, please do not merge on red

**Pull request merged**
